drivers/serial/serial_xen.c

```diff
@@ -40,26 +42,29 @@ int xen_serial_setbrg(struct udevice *dev, int baudrate)
 static int xen_serial_probe(struct udevice *dev)
 {
 	struct xen_uart_priv *priv = dev_get_priv(dev);
-	u64 v = 0;
+	u64 val = 0;
 	unsigned long gfn;
-	int r;
+	int ret;
 
-	r = hvm_get_parameter(HVM_PARAM_CONSOLE_EVTCHN, &v);
-	if (r < 0 || v == 0)
-		return r;
+	ret = hvm_get_parameter(HVM_PARAM_CONSOLE_EVTCHN, &val);
+	if (ret < 0 || val == 0)
+		return ret;
 
-	priv->evtchn = v;
-	console_evtchn = v;
+	priv->evtchn = val;
+	console_evtchn = val;
 
-	r = hvm_get_parameter(HVM_PARAM_CONSOLE_PFN, &v);
-	if (r < 0 || v == 0)
-		return -ENODEV;
+	ret = hvm_get_parameter(HVM_PARAM_CONSOLE_PFN, &val);
+	if (ret < 0)
```

Comment:
val needs to be checked - this is set to a new value in the call above

Reply:
[see here](https://elixir.bootlin.com/linux/v5.8-rc4/source/drivers/tty/hvc/hvc_xen.c#L217)
we need that check